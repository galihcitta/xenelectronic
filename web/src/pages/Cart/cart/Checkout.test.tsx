import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { CheckoutProps, _Checkout } from './Checkout.component';

describe('<Checkout />', () => {
  const mockClearCart = jest.fn();

  const props: CheckoutProps = {
    total: 300,
    clearCart: mockClearCart
  };

  const setup = (props: CheckoutProps) => {
    const utils = render(<_Checkout {...props} />);
    const total = utils.getByText(/total:/i);
    const checkoutBtn = utils.getByText(/checkout/i) as HTMLButtonElement;
    const clearCartBtn = utils.getByText(/clear cart/i) as HTMLButtonElement;

    return { ...utils, total, checkoutBtn, clearCartBtn };
  };

  it('renders correctly', () => {
    const { asFragment } = setup(props);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls `clearCart` action', () => {
    const { checkoutBtn, clearCartBtn, total, rerender } = setup(props);
    fireEvent.click(checkoutBtn);
    fireEvent.click(clearCartBtn);

    expect(mockClearCart).toHaveBeenCalledTimes(2);
    rerender(<_Checkout {...props} total={0} />);

    expect(total).toHaveTextContent('Total: $ 0');
  });
});
