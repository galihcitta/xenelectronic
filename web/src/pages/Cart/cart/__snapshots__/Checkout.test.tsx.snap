// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkout /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa iWsfBx"
  >
    <p
      class="sc-bwzfXH brOaWw"
    >
      Total: $ 300
    </p>
    <button
      class="sc-htpNat hiXDRd"
      type="button"
    >
      Checkout
    </button>
    <button
      class="sc-htpNat hiXDRd"
      type="button"
    >
      Clear Cart
    </button>
  </div>
</DocumentFragment>
`;
